client/src/components/UIElements/SubjectOption.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from "react";
+import "./SubjectOption.scss";
+import { subjectType } from "../../types/subjectType";
+
+interface SubjectOptionProps {
+  name: string;
+  parentId?: string;
+  containMoreSubjects?: boolean;
+  subjects?: subjectType[];
+}
+
+const SubjectOption: React.FC<SubjectOptionProps> = ({
+  name,
+  subjects,
+}) => {
+  const [isHidden, setIsHidden] = useState(true);
+
+  const hideOrShow = () => {
+    setIsHidden(!isHidden);
+  };
```

Comment:
usually fuctions like that are called toggleSomthing, in this case "toggleHidden"